### ISSUE TITLE
TS Server fatal error:  Cannot read property 'path' of undefined

### ISSUE THREAD

**Issue by alexandcote** [microsoft/vscode#95264]
Issue Type: <b>Bug</b>

**TypeScript Version:** 3.7.5
**Steps to reproduce crash**
1. Open a project with a `tsconfig` at the root of the project
2. Create a new file in this project
3. Paste this code in the file
```ts
function example(foo: string) {
    return 'bar';
}
```
4. Don't save the file.

VSCode will automatically detect the file as a typescript file but because the file is not saved, the server will crash.

**TS Server Error Stack**
```
TypeError: Cannot read property 'path' of undefined
    at ProjectService.updateNonInferredProjectFiles (tsserver.js:137473:43)
    at ProjectService.addFilesToNonInferredProject (tsserver.js:137363:22)
    at ProjectService.updateRootAndOptionsOfNonInferredProject (tsserver.js:137511:22)
    at ProjectService.loadConfiguredProject (tsserver.js:137447:22)
    at ProjectService.createAndLoadConfiguredProject (tsserver.js:137387:22)
    at ProjectService.createLoadAndUpdateConfiguredProject (tsserver.js:137392:36)
    at ProjectService.assignProjectToOpenedScriptInfo (tsserver.js:138217:44)
    at tsserver.js:138448:107
    at Object.mapDefined (tsserver.js:565:30)
    at ProjectService.applyChangesInOpenFiles (tsserver.js:138448:48)
    at Session.handlers.ts.createMapFromTemplate._a.<computed> (tsserver.js:139302:46)
    at tsserver.js:140960:88
    at IOSession.Session.executeWithRequestId (tsserver.js:140951:28)
    at IOSession.Session.executeCommand (tsserver.js:140960:33)
    at IOSession.Session.onMessage (tsserver.js:140983:35)
    at Interface.<anonymous> (tsserver.js:142298:27)
    at Interface.emit (events.js:203:13)
    at Interface._onLine (readline.js:316:10)
    at Interface._normalWrite (readline.js:461:12)
    at Socket.ondata (readline.js:172:10)
    at Socket.emit (events.js:203:13)
    at addChunk (_stream_readable.js:295:12)
    at readableAddChunk (_stream_readable.js:276:11)
    at Socket.Readable.push (_stream_readable.js:210:10)
    at Pipe.onStreamRead (internal/stream_base_commons.js:166:17)
```

VS Code version: Code 1.44.1 (a9f8623ec050e5f0b44cc8ce8204a1455884749f, 2020-04-11T01:47:00.296Z)
OS version: Darwin x64 19.4.0

**Comment by mjbvz** [comment]
Do you see this using TypeScript 3.8+?

**Comment by alexandcote** [comment]
@mjbvz, no I don't 😃. With both the VSCode versions or the workspace version.

I tried with Typescript 3.8.3 for the workspace version.

**Comment by mjbvz** [comment]
Thanks for testing.

Based on the steps to reproduce, this seems like a duplicate of https://github.com/microsoft/vscode/issues/85547 which was fixed in TS 3.8. Unfortunately we can't workaround this on the VS Code side, you need to use the newer TS version

Let me know if you see this using TS 3.8+
